### ISSUE TITLE
Cannot import .tar with markdown files in nested folders

### ISSUE THREAD

**Issue by cbop-dev** [zadam/trilium#470]
I've just discovered this app (looks very promising!) and installed the latest version (0.30.7) on Linux (Mint 19.1 Cinnamon), and I've tried importing a tar file containing nothing but markdown notes in several nested folders. It fails every time, with nothing imported, if the tar file contains more than one level of folders. Attached is a "zip" version of a very simple folder--since github doesn't allow .tar files--, with only one subfolder, with two .md notes: one at each level. This simple type of structure fails to import every time for me -- but if I remove the subfolder, it imports fine. In this example, all I have is a tar file with the following files/folders:

```
notes.tar:
   - notes/
       - test2/
           - Test.md
       - Test1.md
```

I've attached the relevant sections of my log file. What seems like the relevant error is: 
```
11:53:53.144 INFO  Could not find existing path notes for notes/test2.
{}
Error: Could not find existing path notes for notes/test2.
    at getParentNoteId (/usr/lib/trilium/resources/app.asar/src/services/import/tar.js:97:23)
    at saveDirectory (/usr/lib/trilium/resources/app.asar/src/services/import/tar.js:184:30)
    at Source.<anonymous> (/usr/lib/trilium/resources/app.asar/src/services/import/tar.js:345:23)
    at Source.emit (events.js:182:13)
    at endReadableNT (/usr/lib/trilium/resources/app.asar/node_modules/tar-stream/node_modules/readable-stream/lib/_stream_readable.js:1055:12)
    at process._tickCallback (internal/process/next_tick.js:63:19) 
```
Seems that it is not creating the second-level folder (here: "test2"), and throwing an error. If I can't batch import my notes, this app is useless to me. (I'm currently a Joplin user.) 

Is this an (easily) fixable bug? Am I doing something wrong? (Can't imagine so: I'm just tar-ing a folder and importing, [per instructions](https://github.com/zadam/trilium/wiki/Markdown).)

[notes.zip](https://github.com/zadam/trilium/files/3026451/notes.zip)
[trilium-fail.log](https://github.com/zadam/trilium/files/3026478/trilium-fail.log)

**Comment by cbop-dev** [comment]
UPDATE: to be clear, this is an error when importing a .tar file _that was not created by using the "export" feature of Trilium_ (the later process generates a .json metadata file that apparently the import process uses to create subfolders.) So, how can one import a tree structure of .md files that was created outside of Trilium? (Like when trying to switch from another note-app to Trilium, such as Joplin.)

I'm trying to assess the app: and one key criterion for measuring the value of an app is how well it handles importing/exporting notes--including external notes--so that one isn't locked-in to the app and can easily switch to/from it. As it appears, Trilium is not able to easily import notes from another app, even if they're just .md files in a folder hierarchy. That would be deal-breaker for me: I'm not going to import one-by-one all my notes and re-categorize them. I need to be able to batch-import them easily. Can this be fixed, or added as a feature?

**Comment by zadam** [comment]
Hello, if I re-pack it into tar then the file imports correctly for me. Could you send me the original tar file at zadam.apps@gmail.com? I suspect this might be caused by some slightly different tar format (unfortunately spec allows for various variations ...)

**Comment by zadam** [comment]
This has been fixed in f00e1235f9cf6ee542912e56c069c29a5206c658

This .tar has an unexpected structure - record for "notes/test2/" directory was declared before "notes/" directory was declared so it seemed like referencing non existing directory. This is quite weird since tar is supposed to be streaming format so you can't just look ahead ...

**Comment by cbop-dev** [comment]
Thanks for the quick fix! I'll look forward to trying out the next release, as the current development version seems a too bit buggy/unstable.
